msticpy/sectools/ip_utils.py

```diff
@@ -0,0 +1,200 @@
+    """Base class for other exceptions."""
+
+
+class DataError(Error):
+    """Raised when thereis a data input error."""
+
+
+def convert_to_ip_entities(ip_str: str) -> Tuple[IpAddress]:
+    """
+    Take in an IP Address string and converts it to an IP Entitity.
+
+    Parameters
+    ----------
+    ip_str : str
+        The string of the IP Address
+
+    Returns
+    -------
+    Tuple
+        The populated IP entities including address and geo-location
+
+    """
+    ip_entities = []
+    if ip_str:
+        if "," in ip_str:
+            addrs = ip_str.split(",")
+        elif " " in ip_str:
+            addrs = ip_str.split(" ")
+        else:
+            addrs = [ip_str]
+
+        for addr in addrs:
+            ip_entity = IpAddress()
+            ip_entity.Address = addr.strip()
+            try:
+                IPLOCATION.lookup_ip(ip_entity=ip_entity)
+            except DataError:
+                pass
+            ip_entities.append(ip_entity)
+    return ip_entities
+
+
+@export
+# pylint: disable=no-else-return
+def get_ip_type(ip_str: str) -> str:
+    """
+    function to validate given value is an IP address and deteremine IPType category
+    (e.g. Private/Public/Multicast).
+
+    Parameters
+    ----------
+    ip_str : str
+        The string of the IP Address
+
+    Returns
+    -------
+    str
+        Returns ip type string using ip address module
+
+    """
+    try:
+        ip.ip_address(ip_str)
+    except ValueError:
+        print(f"{ip_str} does not appear to be an IPv4 or IPv6 address")
+    else:
+        if ip.ip_address(ip_str).is_private:
+            ip_type = "Private"
+        elif ip.ip_address(ip_str).is_multicast:
+            ip_type = "Multicast"
+        elif ip.ip_address(ip_str).is_unspecified:
+            ip_type = "Unspecified"
+        elif ip.ip_address(ip_str).is_reserved:
+            ip_type = "Reserved"
+        elif ip.ip_address(ip_str).is_loopback:
+            ip_type = "Loopback"
+        elif ip.ip_address(ip_str).is_global:
+            ip_type = "Public"
+        elif ip.ip_address(ip_str).is_link_local:
+            ip_type = "Link Local"
+    return ip_type
+
+
+@lru_cache(maxsize=1024)
+# pylint: disable=no-else-return
+def get_whois_info(ip_str: str, show_progress=False) -> Tuple[str, dict]:
+    """
+    Retrieves whois ASN information for given IP address using IPWhois python package.
+
+    Parameters
+    ----------
+    ip_str : str
+        [description]
+    show_progress : bool, optional
+        [description], by default False
+
+    Returns
+    -------
+    IP
+        Details of the IP data collected
+
+    """
+    ip_type = get_ip_type(ip_str)
+    if ip_type == "Public":
+        whois = IPWhois(ip_str)
+        whois_result = whois.lookup_whois()
+        if show_progress:
+            print(".", end="")
+        return whois_result["asn_description"], whois_result
+    else:
+        return "NO ASN Information since IP address is of type", {ip_type}
```

Comment:
This return looks a bit weird. Did you mean to append the ip_type to the error message and return an empty dict {}?

Reply:
f-string typo, should be part of string with ip_type value replaced. I will fix this.